### ISSUE TITLE
harbor 2.4.x upgrade to 2.6.x, database upgrade exception

### ISSUE THREAD

**Issue by kycheng** [goharbor/harbor#19286]
When deploying harbor in k8s, the db pod terminates abnormally for some reason during the upgrade of harbor (the reason is unclear).  The remaining new version directory causes the upgrade to fail.

**Expected behavior and actual behavior:**

It was expected that the database could be upgraded normally, but it actually failed.

**Steps to reproduce the problem:**
1. harbor 2.4.x upgrade to 2.6.x
2. when performing a database upgrade, the pod is forcibly deleted

**Additional context:**

The guess is that the pod was stopped when this code was executed.

https://github.com/goharbor/harbor/blob/77903fafc238c8a6399c8c2f845073af5e07d44a/make/photon/db/docker-entrypoint.sh#L30-L39

After the pod is restarted, the upgrade will check the directory of the new version of the database, and if it exists, the upgrade will not be performed.

https://github.com/goharbor/harbor/blob/77903fafc238c8a6399c8c2f845073af5e07d44a/make/photon/db/docker-entrypoint.sh#L24

But at this time, the data in the new version of the database directory cannot start the database, causing the upgrade to be stuck.

I saw that the new version of Harbor checks whether the database directory of the old version exists to determine whether it needs to be upgraded. Can this measure be applied to 2.6.x?

https://github.com/goharbor/harbor/blob/6d1ad65cf16b9c2fe7bd9949d51afb3f3493fdc6/make/photon/db/docker-entrypoint.sh#L36

**Comment by wy65701436** [comment]
did you have the log of db pod?

**Comment by kycheng** [comment]
> did you have the log of db pod? The above will only be executed whenever the pg version upgrade, however, there is no version change between 2.4 and 2.6.

I am upgrading from 2.2.x to 2.6.x, I know this is not a canonical upgrade path, but this problem may occur during a database upgrade.

I can't provide more logs. I solved the problem by deleting the new version database directory and re-upgrading, but the logs at that time were not kept.

When troubleshooting this problem, it was found that the $PGDATA/pg12 directory already existed, but there was a problem with the data in the directory, and the database could not be started through this directory.

The upgrade program did not continue processing because $PGDATA/pg12 existed.

At that time, the $PGDATA/pg12 program was deleted and the upgrade was performed again, and the problem was solved.

But I don't think this is the end of the matter.
